pkg/controlplane/apiserver/config.go

```diff
@@ -130,7 +132,7 @@ func BuildGenericConfig(
 	kubeClientConfig := genericConfig.LoopbackClientConfig
 	clientgoExternalClient, err := clientgoclientset.NewForConfig(kubeClientConfig)
 	if err != nil {
-		lastErr = fmt.Errorf("failed to create real external clientset: %v", err)
+		lastErr = fmt.Errorf("failed to create real external clientset: %w", err)
```

Comment:
Does it make sense to wrap the error here? Is it expected that the caller may want to unwrap? If not, then `%v` is fine.

I'm not very happy about the linter warning for that. It's only a hint which is supposed to make developers think whether wrapping might be needed from an API perspective, but I suspect that devs are changing it to `%w` only to shut up the linter.

Otherwise the PR looks fine to me.

Reply:
Hard to say whether a consumer will want to unwrap. Isn't it good practice to just use `%w` to not block that? Sounds a little like a philosophical discussion :)

Reply:
It depends on whether the underlying error is supposed to be part of the function's API: https://go.dev/blog/go1.13-errors#whether-to-wrap

I'm fine with doing it here, I was just wondering about your motivation and whether I should do anything about the linter.

/lgtm